packages/twenty-front/src/modules/auth/components/TwoFactorAuthenticationProvisionEffect.tsx

```diff
@@ -0,0 +1,79 @@
+import { useSetRecoilState } from 'recoil';
+import { useEffect } from 'react';
+import { useAuth } from '@/auth/hooks/useAuth';
+import { useSnackBar } from '@/ui/feedback/snack-bar-manager/hooks/useSnackBar';
+import { useLingui } from '@lingui/react/macro';
+import { useNavigateApp } from '~/hooks/useNavigateApp';
+import { AppPath } from '@/types/AppPath';
+import { useReadCaptchaToken } from '@/captcha/hooks/useReadCaptchaToken';
+import { useOrigin } from '@/domain-manager/hooks/useOrigin';
+import { getLoginToken } from '@/apollo/utils/getLoginToken';
+import { qrCodeState } from '@/auth/states/qrCode';
+
+export const TwoFactorAuthenticationSetupEffect = () => {
+  const { initiateOtpProvisioning } = useAuth();
+  const { enqueueErrorSnackBar, enqueueSuccessSnackBar } = useSnackBar();
+
+  const navigate = useNavigateApp();
+  const { readCaptchaToken } = useReadCaptchaToken();
+  const { origin } = useOrigin();
+  const loginToken = getLoginToken();
+  const setQrCodeState = useSetRecoilState(qrCodeState);
+
+  const { t } = useLingui();
+
+  useEffect(() => {
+    const handleTwoFactorAuthenticationProvisioningInitiation = async () => {
+      try {
+        if (!loginToken) {
+          enqueueErrorSnackBar({
+            message: t`Login token missing. Two Factor Authentication setup can not be initiated.`,
+            options: {
+              dedupeKey: 'invalid-session-dedupe-key',
+            },
+          });
+          return navigate(AppPath.SignInUp);
+        }
+
+        const token = await readCaptchaToken();
+        const initiateOTPProvisioningResult = await initiateOtpProvisioning({
```

Comment:
The captcha isn't being renewed between the two calls, it's used in the first loginTokenCall and therefore is returned as invalidated when calling initiateOTPProvisioning. I'm okay to remove captcha protection for now so that we can move on on this PR and fix the more important issues